Avro parser fails to properly deserialize sketches data

Presently, using the avro parser to index data containing [sketches](http://druid.io/docs/latest/development/extensions-core/datasketches-theta.html) fail with the following error:

```
Caused by: com.yahoo.sketches.SketchesArgumentException: Possible Corruption: Illegal Family ID: 244
	at com.yahoo.sketches.Family.idToFamily(Family.java:195)
	at com.yahoo.sketches.theta.Sketch.wrap(Sketch.java:335)
	at com.yahoo.sketches.theta.Sketch.wrap(Sketch.java:315)
	at com.yahoo.sketches.theta.Sketches.wrapSketch(Sketches.java:86)
	at io.druid.query.aggregation.datasketches.theta.SketchHolder.deserializeFromMemory(SketchHolder.java:237)
	at io.druid.query.aggregation.datasketches.theta.SketchHolder.deserializeFromByteArray(SketchHolder.java:229)
	at io.druid.query.aggregation.datasketches.theta.SketchHolder.deserializeFromBase64EncodedString(SketchHolder.java:224)
	at io.druid.query.aggregation.datasketches.theta.SketchHolder.deserialize(SketchHolder.java:205)
	at io.druid.query.aggregation.datasketches.theta.SketchMergeComplexMetricSerde$1.extractValue(SketchMergeComplexMetricSerde.java:64)
	at io.druid.segment.incremental.IncrementalIndex$1IncrementalIndexInputRowColumnSelectorFactory$1.getObject(IncrementalIndex.java:179)
	at io.druid.query.aggregation.datasketches.theta.SketchAggregator.aggregate(SketchAggregator.java:51)
```

This is caused because the sketches are getting incorrectly deserialized into an arrayish-string like  "[25, 112, 44, 10]" instead of a byte array

--- a/extensions-core/avro-extensions/src/main/java/org/apache/druid/data/input/avro/AvroFlattenerMaker.java
+++ b/extensions-core/avro-extensions/src/main/java/org/apache/druid/data/input/avro/AvroFlattenerMaker.java
@@ -31,7 +31,6 @@ import org.apache.druid.java.util.common.StringUtils;
 import org.apache.druid.java.util.common.parsers.ObjectFlatteners;
 
 import java.nio.ByteBuffer;
-import java.util.Arrays;
 import java.util.EnumSet;
 import java.util.List;
 import java.util.Set;
@@ -105,7 +104,8 @@ public class AvroFlattenerMaker implements ObjectFlatteners.FlattenerMaker<Gener
       if (binaryAsString) {
         return StringUtils.fromUtf8(((ByteBuffer) field).array());
       } else {
-        return Arrays.toString(((ByteBuffer) field).array());
+        return ((ByteBuffer) field).array();
+
       }
     }
     if (field instanceof Utf8) {

--- a/extensions-core/avro-extensions/src/main/java/org/apache/druid/data/input/avro/AvroFlattenerMaker.java
+++ b/extensions-core/avro-extensions/src/main/java/org/apache/druid/data/input/avro/AvroFlattenerMaker.java
@@ -105,7 +105,6 @@ public class AvroFlattenerMaker implements ObjectFlatteners.FlattenerMaker<Gener
         return StringUtils.fromUtf8(((ByteBuffer) field).array());
       } else {
         return ((ByteBuffer) field).array();
-
       }
     }
     if (field instanceof Utf8) {

--- a/extensions-core/avro-extensions/src/test/java/org/apache/druid/data/input/AvroHadoopInputRowParserTest.java
+++ b/extensions-core/avro-extensions/src/test/java/org/apache/druid/data/input/AvroHadoopInputRowParserTest.java
@@ -83,7 +83,7 @@ public class AvroHadoopInputRowParserTest
         AvroHadoopInputRowParser.class
     );
     InputRow inputRow = parser2.parseBatch(record).get(0);
-    assertInputRowCorrect(inputRow, DIMENSIONS);
+    assertInputRowCorrect(inputRow, DIMENSIONS, fromPigAvroStorage);
   }
 
 

--- a/extensions-core/avro-extensions/src/test/java/org/apache/druid/data/input/AvroStreamInputRowParserTest.java
+++ b/extensions-core/avro-extensions/src/test/java/org/apache/druid/data/input/AvroStreamInputRowParserTest.java
@@ -216,7 +216,7 @@ public class AvroStreamInputRowParserTest
 
     InputRow inputRow = parser2.parseBatch(ByteBuffer.wrap(out.toByteArray())).get(0);
 
-    assertInputRowCorrect(inputRow, DIMENSIONS);
+    assertInputRowCorrect(inputRow, DIMENSIONS, false);
   }
 
   @Test
@@ -257,10 +257,10 @@ public class AvroStreamInputRowParserTest
 
     InputRow inputRow = parser2.parseBatch(ByteBuffer.wrap(out.toByteArray())).get(0);
 
-    assertInputRowCorrect(inputRow, DIMENSIONS_SCHEMALESS);
+    assertInputRowCorrect(inputRow, DIMENSIONS_SCHEMALESS, false);
   }
 
-  public static void assertInputRowCorrect(InputRow inputRow, List<String> expectedDimensions)
+  public static void assertInputRowCorrect(InputRow inputRow, List<String> expectedDimensions, boolean isFromPigAvro)
   {
     assertEquals(expectedDimensions, inputRow.getDimensions());
     assertEquals(1543698L, inputRow.getTimestampFromEpoch());
@@ -307,10 +307,12 @@ public class AvroStreamInputRowParserTest
     );
     assertEquals(Collections.singletonList(SOME_UNION_VALUE), inputRow.getDimension("someUnion"));
     assertEquals(Collections.emptyList(), inputRow.getDimension("someNull"));
-    assertEquals(Collections.singletonList(String.valueOf(SOME_FIXED_VALUE)), inputRow.getDimension("someFixed"));
+    if (isFromPigAvro) {
+      assertEquals(String.valueOf(SOME_FIXED_VALUE), Arrays.toString((byte[]) inputRow.getRaw("someFixed")));
+    }
     assertEquals(
-        Collections.singletonList(Arrays.toString(SOME_BYTES_VALUE.array())),
-        inputRow.getDimension("someBytes")
+        Arrays.toString(SOME_BYTES_VALUE.array()),
+        Arrays.toString((byte[]) (inputRow.getRaw("someBytes")))
     );
     assertEquals(Collections.singletonList(String.valueOf(MyEnum.ENUM1)), inputRow.getDimension("someEnum"));
     assertEquals(Collections.singletonList(String.valueOf(SOME_RECORD_VALUE)), inputRow.getDimension("someRecord"));

